doc/source/io.rst

```diff
@@ -1175,6 +1175,80 @@ data that appear in some lines but not others:
     0  1  2   3
     1  4  5   6
     2  8  9  10
+    
+Handling "bad" lines - preserving the data
+''''''''''''''''''''
+    
+To preserve all data, you can specify header ``names`` that are long enough:
```

Comment:
"that are long enough" might come across as that the actual string name elements are long length-wise (also ``names`` is singular, not plural, in the context).  I think you can reword to make it clearer that we're talking about about the array itself being long itself.